packages/core/src/engines/PinchEngine.ts

```diff
@@ -265,6 +265,19 @@ export class PinchEngine extends Engine<'pinch'> {
     state._delta = [(-wheelValues(event)[1] / PINCH_WHEEL_RATIO) * state.offset[0], 0]
     V.addTo(state._movement, state._delta)
 
+    // _movement rolls back to when it passed the bounds.
+    const [ox, oy] = state.overflow
+    const [dx, dy] = state._delta
+    const [dirx, diry] = state._direction
+
+    if ((ox < 0 && dx > 0 && dirx < 0) || (ox > 0 && dx < 0 && dirx > 0)) {
+      state._movement[0] = state._movementBound[0] as number
+    }
+
+    if ((oy < 0 && dy > 0 && diry < 0) || (oy > 0 && dy < 0 && diry > 0)) {
+      state._movement[1] = state._movementBound[1] as number
+    }
```

Comment:
copy-paste from the `WheelEngine`:
https://github.com/pmndrs/use-gesture/blob/15df8ff071481b6368f8c6365807057bff99b0f6/packages/core/src/engines/WheelEngine.ts#L25-L36

I could move this to a shared helper or something if you want.

Reply:
No worries, just did it!